packages/community-cli-plugin/src/utils/getWorkspaceRoot.js

```diff
@@ -0,0 +1,84 @@
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ * @format
+ * @oncall react_native
+ */
+
+const fs = require('fs');
+const micromatch = require('micromatch');
+const path = require('path');
+const yaml = require('yaml');
+
+/**
+ * Get the workspace paths from the path of a potential workspace root.
+ *
+ * This supports:
+ * - [NPM workspaces](https://docs.npmjs.com/cli/v10/using-npm/workspaces)
+ * - [Yarn workspaces](https://yarnpkg.com/features/workspaces)
+ * - [PNPM workspaces](https://pnpm.io/workspaces)
+ */
+function getWorkspacePaths(packagePath /*: string */) /*: string[] */ {
+  const result /*: string[] */ = [];
```

Comment:
This function is pushing into an array of paths to avoid giving a preference to neither NPM/Yarn nor PNPM if someone has both configured. Before it would do an early return as soon as it found a `workspaces` array in the `package.json`.

Reply:
No valid use case, I just wanted the code to be agnostic, but I can look for the `pnpm-workspace.yaml` first 👍